lib/src/pubspec.dart

```diff
@@ -23,6 +23,12 @@ import 'utils.dart';
 final _packageName = new RegExp(
     "^${identifierRegExp.pattern}(\\.${identifierRegExp.pattern})*\$");
 
+/// The default SDK constraint for packages that don't declare one.
+///
+/// This allows 2.0.0 dev versions to make the migration proecss smoother.
+final _defaultSdkConstraint =
+    new VersionConstraint.parse("<2.0.0-dev.infinity");
```

Comment:
I'd like *all* packages to have SDK constraints by the time 3.0.0-dev rolls around, which should happen if we ensure that no constraint-less package is ever considered compatible with 2.0.0.